Data/ByteString/Lazy.hs

```diff
@@ -452,6 +454,12 @@ foldr :: (Word8 -> a -> a) -> a -> ByteString -> a
 foldr k = foldrChunks (flip (S.foldr k))
 {-# INLINE foldr #-}
 
+-- | 'foldr'' is like 'foldr', but strict in the accumulator.
+foldr' :: (Word8 -> a -> a) -> a -> ByteString -> a
+foldr' _ a Empty = a
```

Comment:
Shouldn't the accumulator be forced here?

Reply:
I should really think through it again and add some more strictness checks. Over the past month I learned quite a bit more about strictness of folds and I am now in a position to verify it thoroughly enough that such questions can be answered once and for all.

Reply:
So:

* In the empty case the accumulator _is_ the return value, so it is forced exactly as much as the return value is.
* In the chunk case, the accumulator is passed until the end, where the parameter function is applied to it. It is up to the function whether it needs to be forced — what we must make sure to do in order to avoid space leaks is to force its return value at every step.

Therefore I think that _theoretically_ we should not force the accumulator.